task/mongo-optimization-task.js

```diff
@@ -15,7 +15,14 @@ const ObjectId = require('mongodb').ObjectID;
  * Test timeout is increased to 60sec for the function.
  * */
 async function before(db) {
-    await db.collection('opportunities').createIndex({'initiativeId': 1});
+    await db.collection('opportunities').createIndex({
+        'initiativeId': 1,
+        "contacts.questions.category_id": 1, 
+        "contacts.datePublished": 1
+    });
+    await db.collection('clientCriteria').createIndex({
+        'value': 1
```

Comment:
Compound index can work better here, but you need to re-work $lookup section to make it possible

Reply:
I upgraded lookup section and now time is about 3000 ms